src/app/users/[id]/components/UserInfoDetail.tsx

```diff
@@ -1,19 +1,145 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
 
-import { useState } from 'react'
+import { useState, useEffect, ChangeEvent } from 'react';
+import { callGetUser, callPutUserInfo } from '@/_networks/api/user';
 
-export const UserInfoDetail = () => {
-  const [searchAddressPopup, setSearchAddressPopup] = useState(false)
-  const [address, setAddress] = useState('')
+import Skeleton from 'react-loading-skeleton';
+import 'react-loading-skeleton/dist/skeleton.css';
+
+import {
+  UserResponse,
+  Gender,
+  UserRole,
+  CallPutUpdateUserInfoRequestBody,
+} from '@/_types';
+
+interface UserInfoDetailProps {
+  userId: number;
+}
+
+export const UserInfoDetail = (props: UserInfoDetailProps) => {
+  const { userId } = props;
+  const [searchAddressPopup, setSearchAddressPopup] = useState(false);
+  const { data, isLoading } = useQuery<UserResponse>({
+    queryKey: [userId],
+    queryFn: () => callGetUser(userId),
+  });
+
+  const [updateUserInfoRequestBody, setUpdateUserInfoRequestBody] =
+    useState<CallPutUpdateUserInfoRequestBody>({
+      newAddress: data?.address ?? '',
+      newAddressDetail: data?.addressDetail ?? '',
+      newBirthday: data?.birthday ?? '',
+      newGender: data?.gender ?? Gender.남,
+      newName: data?.name ?? '',
+      newPassword: '',
+      newPhone: data?.phone ?? '',
+    });
+
+  // 오토필
+  useEffect(() => {
+    if (data) {
+      setUpdateUserInfoRequestBody({
+        newAddress: data.address,
+        newAddressDetail: data.addressDetail,
+        newBirthday: data.birthday,
+        newGender: data.gender,
+        newName: data.name,
+        newPassword: '',
+        newPhone: data.phone,
+      });
+    }
+  }, [data]);
+
+  const mutation = useMutation({
+    mutationFn: (requestBody: CallPutUpdateUserInfoRequestBody) =>
+      callPutUserInfo(userId, requestBody),
+  });
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setUpdateUserInfoRequestBody((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const [isShowAlert, setIsShowAlert] = useState(false);
+  const [alertMessage, setAlertMessage] = useState('');
+  const [isSuccessAlert, setIsSuccessAlert] = useState(false);
+
+  const validate = () => {
+    let validateErrorMessage = '';
+    if (!updateUserInfoRequestBody.newPassword) {
+      validateErrorMessage = '비밀번호를 입력해주세요.';
+    }
+
+    if (!updateUserInfoRequestBody.newName) {
+      validateErrorMessage = '이름을 입력해주세요.';
+    }
+
+    if (!updateUserInfoRequestBody.newPhone) {
+      validateErrorMessage = '핸드폰 번호를 입력해주세요.';
+    }
+
+    if (!updateUserInfoRequestBody.newAddressDetail) {
+      validateErrorMessage = '상세주소를 입력해주세요.';
+    }
+
+    if (validateErrorMessage.length > 1) {
+      setAlertMessage(validateErrorMessage);
+      setIsShowAlert(true);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const updateUserInfo = async () => {
+    if (validate() === false) {
+      return;
```

Comment:
요건 그냥 참고로 넣어드리자면 부정문 써도 됩니당!

```suggestion
    if (!validate()) {
      return;
```